### PR TITLE
Prior to install a new ONIE SONiC image, delete all partitions except EFI/ONIE

### PR DESCRIPTION
**Pull request by olivier-singla**
**- What I did**
Updated onie os install image script so it will delete every partitions, except the one 

**- How I did it**

**- How to verify it**
Installed a SONiC image with various partitions already existent.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by mslocrian** [comment]
Is it wise to blow away all the partitions whilst installing an image? I would think that it would be helpful to have multiple images available. At least the current known good image, and the new one that you're hoping to go to. Just curious. At least with other NOS's it's nice to be able to jump between two installs.

**Comment by rajendra-dendukuri** [comment]
> Is it wise to blow away all the partitions whilst installing an image? I would think that it would be helpful to have multiple images available. At least the current known good image, and the new one that you're hoping to go to. Just curious. At least with other NOS's it's nice to be able to jump between two installs.

Good point.

Unfortunately one NOS cannot assume the partition layout and GRUB boot order of existing NOS. It can only  be possible if all the NOS implementations follow a strict rule on partitions and bootloader settings. It is typical that entire disk is used when an NOS is installed. 

User's intention is to use another NOS, so we need to make way either by "ONIE Uninstall" or by deleting existing NOS pro-actively like the proposed patch. 

There can be some NOS implementations where they can allow two copies of same NOS. But SONiC installs only one copy on disk. So this feature can quicken the process of re-installing NOS. It can be a quick format vs clean format scenario.

We can discuss.

File: installer/x86_64/install.sh [hunk 1]
```diff
@@ -206,6 +289,55 @@ create_demo_gpt_partition()
...
+    # Clean up block device that contains ONIE
+    local spec_blk_dev=$(echo $blk_dev | sed -e 's#/dev/##')
+
+    [ -b "/dev/$spec_blk_dev" ] || {
+        echo "Error: Unable to determine block device of ONIE install"
+        exit 1
+    }
+
+    if [ "$firmware" = "bios" ] ; then
+       # When a NOS partition is deleted, the grub entry in MBR will no longer
```
Comment (jleveque) [inline installer/x86_64/install.sh]: Indent inside this `if` statement is only 3 spaces. Please adjust to 4 spaces for consistency.

**Comment by lguohan** [comment]
current onie installation policy is to find if there is already sonic partition, if yes, then replace it. if not, then check if there is available space, if not, then not do install. We do not actively remove other partitions. it is up to user to decide which partition to delete. user has to use onie uninstall to uninstall partitions needed by him. If we change our policy to delete all non-sonic partitions, it seems aggressive, and may not be what user want.

**Comment by lguohan** [comment]
I do not see much discussion on this topic, can we close this PR?

**Pull request closed without merging**